src/v/datalake/datalake_manager.cc

```diff
@@ -33,6 +37,8 @@ datalake_manager::datalake_manager(
   ss::sharded<features::feature_table>* features,
   ss::sharded<coordinator::frontend>* frontend,
   ss::sharded<cloud_io::remote>* cloud_io,
+  ss::sharded<iceberg::filesystem_catalog>* catalog,
```

Comment:
can this be a `catalog` not a particular implementation ?

Reply:
It seems to me that it would be enough to pass a ref to local instance using `ss::sharded_parameter`

Reply:
Will do

Reply:
Done